api/v1alpha1/lvmcluster_webhook.go

```diff
@@ -265,20 +265,25 @@ func (v *lvmClusterValidator) ValidateUpdate(_ context.Context, old, new runtime
 	return warnings, nil
 }
 
-func validateDeviceClassesStillExist(old, new []DeviceClass) error {
-	deviceClassMap := make(map[string]bool)
-
-	for _, deviceClass := range old {
-		deviceClassMap[deviceClass.Name] = true
+// validateDeviceClassRemoval validates that device class removal follows the business rules:
+// 1. Cannot delete the last device class
+// 2. Cannot delete default device class
+func validateDeviceClassRemoval(old, new []DeviceClass) error {
+	if len(new) == 0 {
+		return fmt.Errorf("cannot remove all device classes: at least one device class must remain")
 	}
 
+	newDeviceClassMap := make(map[string]DeviceClass)
 	for _, deviceClass := range new {
-		delete(deviceClassMap, deviceClass.Name)
+		newDeviceClassMap[deviceClass.Name] = deviceClass
 	}
 
-	// if any old device class is removed now
-	if len(deviceClassMap) != 0 {
-		return fmt.Errorf("device classes can not be removed from the LVMCluster once added oldDeviceClasses:%v, newDeviceClasses:%v", old, new)
+	for _, oldDeviceClass := range old {
+		if _, exists := newDeviceClassMap[oldDeviceClass.Name]; !exists {
+			if oldDeviceClass.Default {
+				return fmt.Errorf("cannot delete default device class")
```

Comment:
It'd be helpful to include the deleted default device class name in the error message.